src/GraphQL/SchemaExtensions.cs

```diff
@@ -245,6 +246,52 @@ public static void Run(this ISchemaNodeVisitor visitor, ISchema schema)
                 }
             }
         }
+
+        /// <summary>
+        /// Replaces one scalar in the schema to another with the same name.
+        /// </summary>
+        /// <typeparam name="TSchema">Type of the schema.</typeparam>
+        /// <param name="schema">The schema for which to replace the scalar.</param>
+        /// <param name="scalar">New scalar. The replacement occurs by its name.</param>
+        /// <returns>Reference to the provided <paramref name="schema"/>.</returns>
+        public static TSchema ReplaceScalar<TSchema>(this TSchema schema, ScalarGraphType scalar)
+            where TSchema : ISchema
+        {
+            new ReplaceScalarVisitor(scalar).Run(schema);
+            return schema;
+        }
+
+        private sealed class ReplaceScalarVisitor : BaseSchemaNodeVisitor
+        {
+            private readonly ScalarGraphType _replacement;
+
+            public ReplaceScalarVisitor(ScalarGraphType replacement)
+            {
+                _replacement = replacement ?? throw new ArgumentNullException(nameof(replacement));
+            }
+
+            public override void VisitSchema(ISchema schema) => schema.AllTypes.Dictionary[_replacement.Name] = _replacement;
```

Comment:
@Shane32 Fixed https://github.com/graphql-dotnet/graphql-dotnet/pull/2388#issuecomment-797574002

Reply:
Notice that it was not possible to write this line of code outside of the library because `Dictionary` is internal.

Reply:
This line of code needs to verify that the type being replaced was indeed a scalar type.  Otherwise it would be possible to overwrite an `InputObjectGraphType` reference with a `ScalarGraphType` reference.

Reply:
> Notice that it was not possible to write ...

I know so I wrote special method.

> This line of code needs to verify...

Right. Done.